cabby/model/text/region_prediction/train.py

```diff
@@ -0,0 +1,173 @@
+  '''Funcion for saving model.'''
+
+  if save_path == None:
+    return
+
+  state_dict = {'model_state_dict': model.state_dict(),
+          'valid_loss': valid_loss}
+
+  torch.save(state_dict, save_path)
+  logging.info(f'Model saved to ==> {save_path}')
+
+
+def load_checkpoint(load_path: Text, model:  torch.nn.Module,
+          device: torch.device) -> Dict[Text, Sequence]:
+  '''Funcion for loading model.'''
+
+  if load_path == None:
+    return
+
+  state_dict = torch.load(load_path, map_location=device)
+  logging.info(f'Model loaded from <== {load_path}')
+
+  model.load_state_dict(state_dict['model_state_dict'])
+  return state_dict
+
+
+def save_metrics(save_path: Text,
+         train_loss_list: Sequence[float],
+         valid_loss_list: Sequence[float],
+         global_steps_list: Sequence[int],
+         valid_accuracy_list:  Sequence[float]):
+  '''Funcion for saving results.'''
+
+  if save_path == None:
+    return
+
+  state_dict = {'train_loss_list': train_loss_list,
+          'valid_loss_list': valid_loss_list,
+          'global_steps_list': global_steps_list,
+          'valid_accuracy_list': valid_accuracy_list}
+
+  torch.save(state_dict, save_path)
+  logging.info(f'Results saved to ==> {save_path}')
+
+
+def load_metrics(load_path: Text, device: torch.device) -> Dict[Text, float]:
+  '''Funcion for loading results.'''
+
+  if load_path == None:
+    return
+
+  state_dict = torch.load(load_path, map_location=device)
+  logging.info(f'Results loaded from <== {load_path}')
+
+  return state_dict
+
+
+def train_model(model:  torch.nn.Module,
+        device: torch.device,
+        optimizer: AdamW,
+        file_path: Text,
+        train_loader: DataLoader,
+        valid_loader: DataLoader,
+        eval_every: int,
+        num_epochs: int,
+        best_valid_loss: float = float("Inf")):
+  '''Main funcion for training model.'''
+  # initialize running values
+  running_loss = 0.0
+  valid_running_loss = 0.0
+  global_step = 0
+  valid_accuracy_list = []
+  train_loss_list = []
+  valid_loss_list = []
+  global_steps_list = []
+
+  # Training loop.
+  model.train()
+
+  for epoch in range(num_epochs):
+    for batch in train_loader:
+      optimizer.zero_grad()
+      input_ids = batch['input_ids'].to(device)
+      attention_mask = batch['attention_mask'].to(device)
+      labels = batch['labels'].to(device)
+      loss, logits = model(
+        input_ids, attention_mask=attention_mask, labels=labels)
+      loss.backward()
+      optimizer.step()
+
+      # Update running values.
+      running_loss += loss.item()
+      global_step += 1
+
+      # Evaluation step.
+      if global_step % eval_every == 0:
+        model.eval()
+        total, correct = 0, 0
+
+        # Validation loop.
+        for batch in valid_loader:
+          input_ids = batch['input_ids'].to(device)
+          attention_mask = batch['attention_mask'].to(device)
```

Comment:
do you need a epsilon to avoid NaN when computing the logits with the attention mask?